lib/valkyrie/persistence/memory/query_service.rb

```diff
@@ -27,8 +27,8 @@ def find_all
     # @param model [Class] Class to query for.
     # @return [Array<Valkyrie::Resource>] All objects in the persistence backend
     #   with the given class.
-    def find_all_of_model(model:)
-      cache.values.select do |obj|
+    def find_all_of_model(model:, from_results: find_all)
```

Comment:
I'm a little concerned about this having a different method signature than all the other query services. Sure, it's an optional parameter, but it sets a precedent I'm not sure I'm comfortable with.

Reply:
I can appreciate the concern, and my perspective if so long as it remains optional, it is a reasonable extension. People may have use cases in their adapters that require these optional parameters. This follows the "open for extension principle"

I'm wondering about adding a `**` to swallow additional `kwargs` would be a good approach in updating the interface. The following would be the generally acceptable format.

```ruby
def find_all_of_model(model:, **)
end
```

I have some implementation details on a more thorough testing of kwargs that are required and options: https://github.com/samvera-labs/samvera-nesting_indexer/blob/master/lib/samvera/nesting_indexer/adapters/interface_behavior_spec.rb

Reply:
Adapters aren't really meant to be sub-classed, so I'm trying not to implement use cases which "may" exist. I think we shouldn't do this commit for now.

Reply:
One middle-ground is we could just change `cache.values` to `find_all` and get a similar result. Maybe a baby step in the direction you're going for?

Reply:
@tpendragon I'm not advocating for sub-classing. I'm looking at method signatures of the adapter interface. By adding `**kwargs` to the interface, we end up with a forward flexible method signature.

That said, I'll twak and adjust to use `find_all` instead of `cache.values` to move this PR forward.